include/oneapi/tbb/detail/_flow_graph_body_impl.h

```diff
@@ -218,7 +218,7 @@ template <typename Input, typename Output, typename B>
 class type_to_key_function_body_leaf : public type_to_key_function_body<Input, Output> {
 public:
     type_to_key_function_body_leaf( const B &_body ) : body(_body) { }
-    Output operator()(const Input &i) override { return body(i); }
+    Output operator()(const Input &i) override { return tbb::detail::invoke(body, i); }
```

Comment:
I suppose you intend to exclude use of the invoke from the Output& specialization's operator(), since it would used to return a reference to something in the input message, which would be unsafe. Is that right?

Reply:
Actually, I missed that use-case. I checked for its safety and I don't see any unsafe actions, because our interface forces the body to return constant reference for some object inside the input message and never modifies it. I have updated the implementation and added the test for it.